packages/sdk/wallets/wallet-walletconnect/src/wc-wallet.ts

```
@@ -0,0 +1,151 @@
+    console.log('onSessionProposal', proposal)
+  }
+  onSessionCreated = (session: SessionTypes.Created) => {
+    console.log('onSessionCreated', session)
+  }
+  onSessionUpdated = (session: SessionTypes.Update) => {
+    console.log('onSessionUpdated', session)
+  }
+  onSessionDeleted = (session: SessionTypes.DeleteParams) => {
+    console.log('onSessionDeleted', session)
+  }
+
+  onPairingProposal = (pairing: PairingTypes.Proposal) => {
+    this.pairing = pairing
+  }
+  onPairingCreated = (pairing: PairingTypes.Created) => {
+    console.log('onPairingCreated', pairing)
+  }
+  onPairingUpdated = (pairing: PairingTypes.Update) => {
+    console.log('onPairingUpdated', pairing)
+  }
+  onPairingDeleted = (pairing: PairingTypes.DeleteParams) => {
+    console.log('onPairingDeleted', pairing)
+  }
+
+  private async setupClient() {
+    this.client = await WalletConnect.init(this.options)
+
+    this.client.on(CLIENT_EVENTS.session.proposal, this.onSessionProposal)
+    this.client.on(CLIENT_EVENTS.session.created, this.onSessionCreated)
+    this.client.on(CLIENT_EVENTS.session.updated, this.onSessionUpdated)
+    this.client.on(CLIENT_EVENTS.session.deleted, this.onSessionDeleted)
+
+    this.client.on(CLIENT_EVENTS.pairing.proposal, this.onPairingProposal)
+    this.client.on(CLIENT_EVENTS.pairing.created, this.onPairingCreated)
+    this.client.on(CLIENT_EVENTS.pairing.updated, this.onPairingUpdated)
+    this.client.on(CLIENT_EVENTS.pairing.deleted, this.onPairingDeleted)
+
+    // @ts-ignore no-dangling-promise
+    this.client
+      .connect({
+        metadata: this.metadata,
+        permissions: {
+          blockchain: {
+            // alfajores, mainnet, baklava
+            chains: ['celo:44787', 'celo:42220', 'celo:62320'],
+          },
+          jsonrpc: {
+            methods: Object.values(SupportedMethods),
+          },
+        },
+      })
+      .then((session) => (this.session = session))
```

Recommend using `await`, promise `.then`s swallow errors

---

Yeah this is due to some hoops I'm jumping through with establishing connections. Mentioned on Slack but will have another pass at it to try get a cleaner API